From e441c03955ed4c676aa324a5b88e7670903b4294 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 3 Apr 2024 14:39:21 +0200
Subject: [PATCH 0747/1695] fdts: stm32: describe cpu opp for STM32MP2

Describe CPU operating points in the STM32MP2 device trees.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ddaf622115b8bb1a5bb699aa7a2d28ae54c5faa4 fdts: stm32: describe cpu opp for STM32MP2]
Change-Id: I89b5a61b2ff0032d4f384ac993bd07c4a624607d
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/371770
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi     | 19 +++++++++++++++++++
 core/arch/arm/dts/stm32mp257f-dk.dts  | 23 +++++++++++++++++++++++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 23 +++++++++++++++++++++++
 3 files changed, 65 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 769ad7227..6dca04d18 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -23,6 +23,25 @@
 			device_type = "cpu";
 			reg = <0>;
 			enable-method = "psci";
+			operating-points-v2 = <&cpu0_opp_table>;
+			clocks = <&rcc CK_CPU1>;
+		};
+	};
+
+	cpu0_opp_table: cpu0-opp-table {
+		compatible = "operating-points-v2";
+
+		opp-1200000000 {
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <800000>;
+			opp-supported-hw = <0x3>;
+			st,opp-default;
+		};
+
+		opp-1500000000 {
+			opp-hz = /bits/ 64 <1500000000>;
+			opp-microvolt = <910000>;
+			opp-supported-hw = <0x2>;
 		};
 	};
 
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index fef73e674..31d0cdfd8 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -48,6 +48,10 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&vddcpu>;
+};
+
 &hash {
 	status = "okay";
 };
@@ -379,6 +383,25 @@
 	};
 };
 
+&rcc {
+	st,c1msrd = <2>;
+	st,clk_opp {
+		st,ck_cpu1 {
+			cfg_1 {
+				hz = <1500000000>;
+				st,clksrc = <0>;
+				st,pll = <&pll1_cfg_1500Mhz>;
+			};
+
+			cfg_2 {
+				hz = <1200000000>;
+				st,clksrc = <0>;
+				st,pll = <&pll1_cfg_1200Mhz>;
+			};
+		};
+	};
+};
+
 &risaf2 {
 	status = "okay";
 };
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 76a48331d..0dcf40ef9 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -48,6 +48,10 @@
 	};
 };
 
+&cpu0 {
+	cpu-supply = <&vddcpu>;
+};
+
 &hash {
 	status = "okay";
 };
@@ -414,6 +418,25 @@
 	};
 };
 
+&rcc {
+	st,c1msrd = <2>;
+	st,clk_opp {
+		st,ck_cpu1 {
+			cfg_1 {
+				hz = <1500000000>;
+				st,clksrc = <0>;
+				st,pll = <&pll1_cfg_1500Mhz>;
+			};
+
+			cfg_2 {
+				hz = <1200000000>;
+				st,clksrc = <0>;
+				st,pll = <&pll1_cfg_1200Mhz>;
+			};
+		};
+	};
+};
+
 &risaf2 {
 	status = "okay";
 };
-- 
2.47.2

